### PR TITLE
punto14

### PR DESCRIPTION
**Pull request by jeffrey20122021**

### CODE REVIEW

File: cypress/page/forms.page.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      this.submit = "#submit";
+      this.formURL = "https://demoqa.com/automation-practice-form";
+      this.modal = "#example-modal-sizes-title-lg";
+    }
+  
+    public visitUrlPage(): void {
+      cy.visit(this.formURL);
+    }
+  
+    public fillForm(Info: PersonalInfo): void {
```
Comment (holgiosalos) [inline cypress/page/forms.page.ts:36]: parameters on methods should be in camelCase, so rename `Info` as `info` or `personalInfo`

File: cypress/e2e/fill-form.cy.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    hobbies: ["Music", "Reading"],
+    currentAddress: "Av siempreViva # 123",
+    state: "Uttar Pradesh",
+    city: "Agra",
+  };
+  before(()=> {
+    personalFormPage = new PersonalFormPage();
+  });
+
+  it(() => {
```
Comment (holgiosalos) [inline cypress/e2e/fill-form.cy.ts:21]: please give a description to the test

**Review by holgiosalos — COMMENTED** [review]
Apply this code to avoid your tests failing:
image

File: cypress/e2e/fill-form.cy.ts (hunk 2 above)
Comment (jeffrey20122021) [reply cypress/e2e/fill-form.cy.ts]: all done @holgiosalos

File: cypress/page/forms.page.ts [hunk 3]
```diff
@@ -33,20 +33,20 @@ class PersonalFormPage {
...
-      Info.hobbies.forEach((hobbie) => {
+    public fillForm(info: PersonalInfo): void {
+      cy.get(this.firstName).type(info.name);
+      cy.get(this.lastName).type(info.lastName);
+      cy.get(this.email).type(info.email);
+      cy.get(this.gender).find(`input[value = '${info.gender}']`).click({force: true});
+      cy.get(this.mobile).type(info.mobileNumber);
+      cy.get(this.birthDate).type(`{selectall}${info.dateOfBirth}{enter}`);
+      info.hobbies.forEach((hobbie) => {
         cy.get(this.hobbies).find(".custom-control-label").filter(`:contains("${hobbie}")`).click();
```
Comment (holgiosalos) [inline cypress/page/forms.page.ts:44]: is the find needed? if so, you can try to modify the `this.hobbies` property to have this value `#hobbiesWrapper input+label`, in that case we are telling cypress to look in the labels

**Review by holgiosalos — COMMENTED** [review]
remember to run  the lint:fix command to avoid fixable errors appear in the CI

File: cypress/page/forms.page.ts (hunk 3 above)
Comment (jeffrey20122021) [reply cypress/page/forms.page.ts]: The find is no neccesary . With this.hobbies is selecting the #hobbiesWrapper ID and then filter the hobbies.

**Review by holgiosalos — APPROVED** [review]

**Pull request merged**